### PR TITLE
feat: Use docker for reproducible canister builds

### PR DESCRIPTION
**Pull request by dsarlis**
Add a Dockerfile that allows for reproducible canister builds for both the bitcoin as well as the uploader canister. To share most of the parts, the build script for the bitcoin canister is moved to `scripts/build-canister.sh` that accepts as an argument the name of the canister that should be built.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#
+# docker build -t canisters .
+# docker run --rm --entrypoint cat canisters /ic-btc-canister.wasm > ic-btc-canister.wasm
+# docker run --rm --entrypoint cat canisters /uploader-canister.wasm > uploader-canister.wasm
+
+# The docker image. To update, run `docker pull ubuntu` locally, and update the
+# sha256:... accordingly.
+FROM ubuntu@sha256:626ffe58f6e7566e00254b638eb7e0f3b11d4da9675088f4781a50ae288f3322
+
+ARG rust_version=1.62.1
```
Comment (ielashi) [inline Dockerfile:11]: nit: can we add a note here that, if we update this version, we should also update the version in `rust-toolchain.toml`? And vice versa in the `rust-toolchain.toml` file as well.

**Review by ielashi — APPROVED** [review]
LGTM, thanks!

**Comment by ielashi** [comment]
@dsarlis Also, please make sure you're merging this commit with your DFINITY email address, not your personal one.

**Comment by dsarlis** [comment]
> @dsarlis Also, please make sure you're merging this commit with your DFINITY email address, not your personal one.

Yeah, I'm using my dfinity address.

File: Dockerfile (hunk 1 above)
Comment (dsarlis) [reply Dockerfile:13]: Done.

**Pull request merged**